build/rpmfc.c

```diff
@@ -514,6 +514,22 @@ static ARGV_t runCmd(const char *cmd,
     return output;
 }
 
+static ARGV_t runCall(const char *cmd,
+		     const char *buildRoot, const char *fn)
+{
+
+    ARGV_t output = NULL;
+    char *fullcmd = rstrscat(NULL, "%{", cmd, " ", "%{?buildroot:buildroot/}", fn, "}", NULL);
```

Comment:
This doesn't work for file names like `foo}bar` or `foo%%bar`. To make that bulletproof, you'd need a `rpmMacroExpandParametric(rpmMacroContext mc, const char *n, ARGV_t argv)` function.

Reply:
Hmm, annoying but true. One simple way around this would be passing the argument via a separately defined macro, eg
```
    char *path = rstrscat(NULL, buildRoot ? buildRoot : "", "/", fn, NULL);
    rpmPushMacro(NULL, "filename", NULL, path, 0);
    char *exp = rpmExpand("%{", cmd, "}", NULL);
    rpmPopMacro(NULL, "filename");
    ...
```

...and then use %{filename} instead of %{1} in the macro, but that's not very elegant. Maybe there should be a rpmPushMacroArg() or such to get around this in a generic way.

Reply:
Hmm, actually just pushing/popping a macro named "1" gets around that. It's a bit of a cheat of course but it'll probably do for now.

Reply:
Fixed in the latest push by faking the %1 argument.

Reply:
I'm not sure but I think you still need to double the '%' chars...

Reply:
Possibly. I don't know if I can bring myself to care enough about that fringe case though...

Reply:
The latest version using literal resolves even the % case.